ports/stm32/boards/ARDUINO_PORTENTA_H7/mpconfigboard.h

```diff
@@ -0,0 +1,326 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ * The MIT License (MIT)
+ * Copyright (c) 2022 Arduino SA
+ */
+
+#define MICROPY_HW_BOARD_NAME       "PORTENTA"
+#define MICROPY_HW_MCU_NAME         "STM32H747"
+#define MICROPY_PY_SYS_PLATFORM     "Portenta"
+#define MICROPY_HW_FLASH_FS_LABEL	"portenta"
+#define MICROPY_PY_SYS_STDIO_BUFFER (0)
+#define MICROPY_STREAMS_POSIX_API   (1)
+
+#define MICROPY_OBJ_REPR            (MICROPY_OBJ_REPR_C)
```

Comment:
Why did you go with repr C?

Reply:
Mainly because `REPR_A` allocates an `mp_obj_float_t` object per float, Vs. `REPR_C` in which a float is just 4 bytes. This is an issue if you need to allocate float arrays, they will take a lot of RAM. Note I want this configuration to be usable by our firmware too, so I don't have to maintain another `mpconfigboard.h`. If there's an easy option to override that when building ex `-DMICROPY_PY_REPR_C=1` we could use `REPR_A` here.

Reply:
> This is an issue if you need to allocate float arrays, they will take a lot of RAM

`array.array('f', ...)` is still just 4-bytes per float, regardless of the object representation.  It's only individual float objects that change when the repr changes; eg from `x = array[0]`, `x` would take 16 bytes on the heap in repr A, but on repr C the value is stored directly in the object reference, like a small int.  Repr C allows float operations without using the heap, similar to repr D.

I'm OK with having repr C configured here, it's just good to know why it's chosen, and maybe worth a comment.

Reply:
I meant float arrays returned from C functions, for example a function that gets called in a loop and returns this

```c
    mp_obj_t tuple[3];
    tuple[0] = mp_obj_new_float(Ta);
    tuple[1] = mp_obj_new_float(min);
    tuple[2] = mp_obj_new_float(max);
    return mp_obj_new_tuple(3, tuple);
```

I suppose there might be a way to create that array from C and that it's mutable ?

Reply:
Right, so that's a tuple of floats.  Same would go for a list of floats.  Both do allocate the float values on the heap in repr A.

You can create an `array.array` from C but there are no convenience functions for that.  I'm happy to make one (eg `mp_obj_new_array(type, size)`) if that's helpful?

Reply:
We use both a tuple or a list of floats, for example to return an image float pixels. While that function would be very helpful, we also commonly use `mp_obj_new_float` to return single floats, each would require an alloc, so `REPR_C` is still much less overhead I think.

Reply:
Yes REPR_C is much less overhead.  But keep in mind it's a 30-bit FP number, not 32-bit.

Reply:
Yes that's an acceptable tradeoff some precision for RAM.